src/lib/tasks/server/send_query_response_task.cpp

```diff
@@ -0,0 +1,121 @@
+#include "send_query_response_task.hpp"
+
+#include "server/postgres_wire_handler.hpp"
+#include "sql/sql_pipeline.hpp"
+
+#include "SQLParserResult.h"
+
+namespace opossum {
+
+std::vector<ColumnDescription> SendQueryResponseTask::build_row_description(const std::shared_ptr<const Table> table) {
+  std::vector<ColumnDescription> result;
+
+  const auto& column_names = table->column_names();
+  const auto& column_types = table->column_types();
+
+  for (auto column_id = 0u; column_id < table->column_count(); ++column_id) {
+    switch (column_types[column_id]) {
+      case DataType::Int:
+        result.emplace_back(ColumnDescription{column_names[column_id], /* object_id */ 23, /* type id */ 4});
+        break;
+      case DataType::Long:
+        result.emplace_back(ColumnDescription{column_names[column_id], /* object_id */ 20, /* type id */ 8});
+        break;
+      case DataType::Float:
+        result.emplace_back(ColumnDescription{column_names[column_id], /* object_id */ 700, /* type id */ 4});
+        break;
+      case DataType::Double:
+        result.emplace_back(ColumnDescription{column_names[column_id], /* object_id */ 701, /* type id */ 8});
+        break;
+      case DataType::String:
+        result.emplace_back(ColumnDescription{column_names[column_id], /* object_id */ 25, /* type id */ -1});
+        break;
+      default:
+        Fail("Bad DataType");
+    }
+  }
+
+  return result;
+}
+
+std::string SendQueryResponseTask::build_command_complete_message(hsql::StatementType statement_type,
+                                                                  uint64_t row_count) {
+  switch (statement_type) {
+    case hsql::StatementType::kStmtSelect: {
+      return "SELECT " + std::to_string(row_count);
+    }
+    case hsql::StatementType::kStmtInsert: {
+      // 0 is ignored OID and 1 inserted row
+      return "INSERT 0 1";
+    }
+    case hsql::StatementType::kStmtUpdate: {
+      // We do not return how many rows are affected
```

Comment:
Why?

Reply:
Because we don't track that information right now. But could we return `-1` or any other number so that people don't have to wonder why nothing happened?